nerfstudio/data/datamanagers/full_images_datamanager.py

```diff
@@ -180,26 +180,30 @@ def process_eval_data(idx):
             return data
 
         CONSOLE.log("Caching / undistorting train images")
-        with ThreadPoolExecutor() as executor:
+        with ThreadPoolExecutor(max_workers=2) as executor:
             cached_train = list(
-                executor.map(
-                    process_train_data,
-                    track(
+                track(
+                    executor.map(
+                        process_train_data,
                         range(len(self.train_dataset)),
-                        description="Caching / undistorting train images",
-                        transient=True,
                     ),
+                    description="Caching / undistorting train images",
+                    transient=True,
+                    total=len(self.train_dataset),
                 )
             )
 
         CONSOLE.log("Caching / undistorting eval images")
-        with ThreadPoolExecutor() as executor:
+        with ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
please make this configurable? or maybe this is just for debugging

Reply:
I'm actually not sure why this shows up in this diff, the change is from https://github.com/nerfstudio-project/nerfstudio/pull/2969. it speeds up undistortion a lot for big datasets I've been toying with!

it's hardcoded to 2 because we can really only expect benefits from one thread doing IO while the other thread is doing undistortion; the implementation is still weird given this (ideally we'd just have 1 worker doing IO while the main one is sequentially undistorting) but I'm a fan of not letting perfect be the enemy of... better 🤷

Reply:
FWIW each worker might need `cv2.setNumThreads(1)` or else it can choke the CPU.  I seem to see way more than 200% util here hence why i brought it up so maybe it's just not tuned well for all users.

maybe in a future refactor this stuff will just get pushed to a torch dataloader... the pinned memory part breaks for me for large datasets anyways, literally I got a OOM and hard lock-up because too too too much much much pinned memory